File: java/src/main/java/io/hatake/core/ValidatorDefinition.java
package io.hatake.core;

import java.util.Map;

/**
 * A validation rule. {@code type} selects the validator; {@code params} carries
 * its arguments (e.g. {@code {"value": 20}} for maxLength). {@code message}
 * optionally overrides the default message.
 */
public record ValidatorDefinition(String type, Map<String, Object> params, String message) {
}

File: flutter/packages/hatake_core/lib/src/definition/validator_definition.dart
import 'package:equatable/equatable.dart';

/// Declares a validation rule to apply to a field.
///
/// [type] is an open string (see `ValidatorTypes`); [params] carries rule
/// arguments (e.g. `{'value': 20}` for `maxLength`). Execution is performed by
/// a validator registered against [type], not by this definition.
class ValidatorDefinition extends Equatable {
  final String type;
  final Map<String, Object?> params;

  /// Optional override message shown when validation fails.
  final String? message;

  const ValidatorDefinition({
    required this.type,
    this.params = const {},
    this.message,
  });

  @override
  List<Object?> get props => [type, params, message];
}
